src/story.ts

```diff
@@ -0,0 +1,31 @@
+export type StoryId = string;
+export type StoryKind = string;
+export type StoryName = string;
+
+type StoryReturnType<T> = T extends (storyFn: infer StoryFn, context: any) => any
+  ? StoryFn extends (...args: any) => any
+    ? ReturnType<StoryFn>
+    : never
+  : never;
+
+export interface KindMeta<Component = unknown, Decorator = unknown> {
+  id?: StoryId;
+  title: StoryKind;
+  component?: Component;
+  subcomponents?: unknown; // TODO
+  decorators?: Decorator[];
+  parameters?: {
+    [name: string]: unknown;
+  };
+}
+
+export interface StoryMeta<Decorator = unknown> {
+  (): StoryReturnType<Decorator>;
+  story?: {
+    name?: StoryName;
+    decorators?: Decorator[];
+    parameters?: {
```

Comment:
DRY: `Parameters` here and above

Reply:
when to use `any` vs `unknown`?

Reply:
`unknown` is more strict. For example, you could define and use any props in parameters, but when you want to do something with prop value, like call method or take internal prop or do arithmetic operation, typescript warn you about that the type is unknown, so you need explicitly cast type or extends parameters type through generic.